### DISCUSSION TITLE
Asterisks for Date/Time of Identical Files??

### DISCUSSION THREAD

**Discussion by Hammerhead10002** [WinMerge/winmerge#766] [Q&A]
Hello All,
I've been using WinMerge for several years and really like it. I have run up against something simple (?) that I do not understand, and I can not find an answer in Help or on the Net. 

Why do I often (but not always) see an Asterisk in front of the Modified Date for "Identical files"? 
- I am currently using v2.16.8.0, but have noticed this with all past versions as well. (Yes, I will update my version).
- Help tells me "The newer of the two dates (or the date for any unique item) is prefixed with an asterisk ( * )." - These files are identical and not unique (Unless I am missing something!).
- I have read that WinMerge does not comprehend "original", so I don't believe that is an issue.
- I am duplicating some large folders with subfolders using Teracopy set to duplicate attributes and date.
- Teracopy verifies the files as identical using CRC/MD5.

Please see the attached screen shot. "1" shows asterisk on identical files. "2" shows no asterisk on identical files.

Any help would be greatly appreciated!
![WM_03c](https://user-images.githubusercontent.com/83463874/116720750-8e305a00-a9aa-11eb-8316-a3986834adc5.jpg)

**Comment by sdottaka** [comment]
Perhaps the timestamps of the files match up to the second, but the millisecond values that are hidden are different.

If you run the following in PowerShell to show the milliseconds or less of the modification times of the two files, how does it look?

~~~
 (dir filepath).LastWriteTime.ToString("yyyy-MM-dd HH:mm:ss.fffffff")
~~~

**Reply by Hammerhead10002** [reply]
Hello sdottaka ... thanks so much for the help! I am a complete beginner with Powershell... in fact this was my first time using it for a definite purpose. I couldn't get that script to run (errors), and I don't know enough to understand what I was doing wrong. I was, however, able to get the following script to run, and I think it brought me to the intended result (?)  
Example-  (Get-Item O:\320BU2).LastWriteTime.toString("yyyy-MM-dd HH:mm:ss.fffffff").  

The results:
Directory with no asterisk:
![20210501_WM_Snip_07a](https://user-images.githubusercontent.com/83463874/116788449-9b674a80-aa77-11eb-9970-1161ae85ddd2.JPG)
![20210501_WM_Snip_07b](https://user-images.githubusercontent.com/83463874/116788450-a4581c00-aa77-11eb-89fd-a50b382600b0.JPG)

Directory with asterisk #1
![20210501_WM_Snip_05a](https://user-images.githubusercontent.com/83463874/116788496-dff2e600-aa77-11eb-8d87-5c19942f912f.JPG)
![20210501_WM_Snip_05b](https://user-images.githubusercontent.com/83463874/116788498-e4b79a00-aa77-11eb-9814-1cffdaf6b54c.JPG)

Directory with asterisk #2 
![20210501_WM_Snip_06a](https://user-images.githubusercontent.com/83463874/116788516-044ec280-aa78-11eb-9e11-c963fe61141d.JPG)
![20210501_WM_Snip_06b](https://user-images.githubusercontent.com/83463874/116788519-087ae000-aa78-11eb-92e4-bb23e8c11f2f.JPG)

I can see that the directory/file pairs with no asterisk have identical extended seconds, and the directory/file pairs with one asterisk show the newer of the two (even if it is microseconds)! I didn't know that WinMerge operates to this "Time resolution". I have been wondering about that for years! :)

Brief questions, please:
- It is relatively safe to assume this is the cause of the asterisks? The directories/files should otherwise be identical?
- What would cause the "micro" delays on only some of the directories/files? Processing times and loads?

Thanks so much for the help. It looks like you were right on with this! 👍
